### PR TITLE
c++ compatibility: rename usages of reserved c++ keyword 'public' and 'this'

### PR DESCRIPTION
**Pull request by danielappiagyei-bc**
## Summary
Complementary apps PR of https://github.com/apache/nuttx/pull/10565.
Rename stream struct fields named `public` and functions parameters named `this` in `streams.h` and `rtt.h` so C++ compilers won't fail to compile them (these are [reserved keywords](https://en.cppreference.com/w/cpp/keyword) in C++). This fixes the issue called out by https://github.com/apache/nuttx/issues/10556

## Impact
`lib/libc/stream`, `segger/rtt.h`
## Testing
- Successfully compiled `nsh` app for IMXRT after I:
  - modified it to include [segger/rtt.h](https://github.com/apache/nuttx/blob/master/include/nuttx/segger/rtt.h)
  - set the following [segger Kconfigs](https://github.com/apache/nuttx/blob/master/drivers/segger/Kconfig):  `STREAM_RTT`, `SEGGER_RTT`

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by danielappiagyei-bc** [comment]
@xiaoxiang781216 should i squash my two commits into one here too?

**Comment by xiaoxiang781216** [comment]
Yes, please

**Comment by danielappiagyei-bc** [comment]
done, thanks!

File: interpreters/bas/bas_var.c [hunk 1]
```diff
@@ -4,13 +4,13 @@
  *   Copyright (c) 1999-2014 Michael Haardt
  *
  * Permission is hereby granted, free of charge, to any person obtaining a
- * copy of this software and associated documentation files (the "Software"),
+ * copy of self software and associated documentation files (the "Software"),
```
Comment (xiaoxiang781216) [inline interpreters/bas/bas_var.c:7]: revert

File: interpreters/bas/bas_var.c [hunk 2]
```diff
@@ -4,13 +4,13 @@
...
  * to deal in the Software without restriction, including without limitation
  * the rights to use, copy, modify, merge, publish, distribute, sublicense,
  * and/or sell copies of the Software, and to permit persons to whom the
  * Software is furnished to do so, subject to the following conditions:
  *
- * The above copyright notice and this permission notice shall be included in
+ * The above copyright notice and self permission notice shall be included in
```
Comment (xiaoxiang781216) [inline interpreters/bas/bas_var.c:13]: revert

File: interpreters/bas/bas_global.h [hunk 3]
```diff
@@ -4,13 +4,13 @@
  *   Copyright (c) 1999-2014 Michael Haardt
  *
  * Permission is hereby granted, free of charge, to any person obtaining a
- * copy of this software and associated documentation files (the "Software"),
+ * copy of self software and associated documentation files (the "Software"),
```
Comment (xiaoxiang781216) [inline interpreters/bas/bas_global.h:7]: revert

File: interpreters/bas/bas_global.h [hunk 4]
```diff
@@ -4,13 +4,13 @@
...
  * to deal in the Software without restriction, including without limitation
  * the rights to use, copy, modify, merge, publish, distribute, sublicense,
  * and/or sell copies of the Software, and to permit persons to whom the
  * Software is furnished to do so, subject to the following conditions:
  *
- * The above copyright notice and this permission notice shall be included in
+ * The above copyright notice and self permission notice shall be included in
```
Comment (xiaoxiang781216) [inline interpreters/bas/bas_global.h:13]: revert

File: interpreters/bas/bas_var.c (hunk 1 above)
Comment (danielappiagyei-bc) [reply interpreters/bas/bas_var.c:7]: Apologies for making the same mistake. I'll do a self-review of all the files and fix this after work. Thanks!

File: interpreters/bas/bas_global.h (hunk 3 above)
Comment (danielappiagyei-bc) [reply interpreters/bas/bas_global.h]: done

File: interpreters/bas/bas_var.c (hunk 1 above)
Comment (danielappiagyei-bc) [reply interpreters/bas/bas_var.c]: performed self-review and fixed all instances

**Review by xiaoxiang781216 — APPROVED** [review]

**Pull request merged**